### ISSUE TITLE
Allow `modules` in a `group`

### ISSUE THREAD

**Issue by parsonsmatt** [ndmitchell/hlint#1639]
We want to group rules under teams for which modules can be imported and where.

```yaml
- group:
    name: Some Cool Team Rules
    rules:
      - modules: ...
```

However, this fails to parse: requires `error`, `warning`, key. Nesting it under `error` does not work.
